### PR TITLE
util/arena: migrate test-infra to testify

### LINKED ISSUE #26103: migrate test-infra to testify for `util/arena` pkg

### PR DESCRIPTION
**Pull request by Ranxy**
### What problem does this PR solve?
Issue Number: close #26103

### What's Changed:

Migrate test-infra to testify for util/arena pkg.

### Release note
* No release note

### CODE REVIEW

**Comment by tisonkun** [comment]
@Ranxy the next time plz comment under the original issue first so that others won't conflict with you.

**Comment by Ranxy** [comment]
@tisonkun Okay, I'll keep an eye on this next time.

File: util/arena/arena_test.go [hunk 1]
```diff
@@ -16,73 +16,37 @@ package arena
...
 	slice := arena.Alloc(10)
-	if arena.off != 10 {
-		t.Error("off not match, expect 10 bug got", arena.off)
-	}
-
-	if len(slice) != 0 || cap(slice) != 10 {
-		t.Error("slice length or cap not match")
-	}
+	assert.Equal(t, 10, arena.off, "off not match, expect 10 bug got")
+	assert.False(t, len(slice) != 0 || cap(slice) != 10, "slice length or cap not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:26]: We may use two positive assertions here.

```suggestion
	assert.Len(t, slice, 0)
	assert.Equal(t, 10, cap(slice))
```

Also the comment is redundant, `testify` will generate pretty error message.

File: util/arena/arena_test.go [hunk 2]
```diff
@@ -16,73 +16,37 @@ package arena
...
 	arena := NewAllocator(1000)
 	slice := arena.Alloc(10)
-	if arena.off != 10 {
-		t.Error("off not match, expect 10 bug got", arena.off)
-	}
-
-	if len(slice) != 0 || cap(slice) != 10 {
-		t.Error("slice length or cap not match")
-	}
+	assert.Equal(t, 10, arena.off, "off not match, expect 10 bug got")
```
Comment (tisonkun) [inline util/arena/arena_test.go:25]:
```suggestion
	assert.Equal(t, 10, arena.off)
```

ditto redundant comment.

File: util/arena/arena_test.go [hunk 3]
```diff
@@ -16,73 +16,37 @@ package arena
...
 
 	slice = arena.Alloc(20)
-	if arena.off != 30 {
-		t.Error("off not match, expect 30 bug got", arena.off)
-	}
-
-	if len(slice) != 0 || cap(slice) != 20 {
-		t.Error("slice length or cap not match")
-	}
+	assert.Equal(t, 30, arena.off, "off not match, expect 30 bug got")
```
Comment (tisonkun) [inline util/arena/arena_test.go:29]:
```suggestion
	assert.Equal(t, 30, arena.off)
```

ditto redundant comment, also the info "30" is duplicate, what if it changes later?

File: util/arena/arena_test.go [hunk 4]
```diff
@@ -16,73 +16,37 @@ package arena
...
+	assert.False(t, len(slice) != 0 || cap(slice) != 20, "slice length or cap not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:30]:
```suggestion
	assert.Len(t, slice, 0)
	assert.Equal(t, 20, cap(slice))
```

ditto better assertions.

File: util/arena/arena_test.go [hunk 5]
```diff
@@ -16,73 +16,37 @@ package arena
...
 
 	slice = arena.Alloc(1024)
-	if arena.off != 30 {
-		t.Error("off not match, expect 30 bug got", arena.off)
-	}
-
-	if len(slice) != 0 || cap(slice) != 1024 {
-		t.Error("slice length or cap not match")
-	}
+	assert.Equal(t, 30, arena.off, "off not match, expect 30 bug got")
```
Comment (tisonkun) [inline util/arena/arena_test.go:33]:
```suggestion
	assert.Equal(t, 30, arena.off)
```

File: util/arena/arena_test.go [hunk 6]
```diff
@@ -16,73 +16,37 @@ package arena
...
+	assert.False(t, len(slice) != 0 || cap(slice) != 1024, "slice length or cap not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:34]:
```suggestion
	assert.Len(t, slice, 0)
	assert.Equal(t, 1024, cap(slice))
```

File: util/arena/arena_test.go [hunk 7]
```diff
@@ -16,73 +16,37 @@ package arena
...
 
 	slice = arena.AllocWithLen(2, 10)
-	if arena.off != 40 {
-		t.Error("off not match, expect 40 bug got", arena.off)
-	}
-
-	if len(slice) != 2 || cap(slice) != 10 {
-		t.Error("slice length or cap not match")
-	}
+	assert.Equal(t, 40, arena.off, "off not match, expect 40 bug got")
```
Comment (tisonkun) [inline util/arena/arena_test.go:37]:
```suggestion
	assert.Equal(t, 40, arena.off)
```

File: util/arena/arena_test.go [hunk 8]
```diff
@@ -16,73 +16,37 @@ package arena
...
+	assert.False(t, len(slice) != 2 || cap(slice) != 10, "slice length or cap not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:38]:
```suggestion
	assert.Len(t, slice, 2)
	assert.Equal(10, cap(slice))
```

File: util/arena/arena_test.go [hunk 9]
```diff
@@ -16,73 +16,37 @@ package arena
...
 
 	arena.Reset()
-	if arena.off != 0 || cap(arena.arena) != 1000 {
-		t.Error("off or cap not match")
-	}
+	assert.False(t, arena.off != 0 || cap(arena.arena) != 1000, "off or cap not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:41]:
```suggestion
	assert.Equal(t, 0, arena.off)
	assert.Equal(t, 1000, cap(arena.arena))
```

File: util/arena/arena_test.go [hunk 10]
```diff
@@ -16,73 +16,37 @@ package arena
...
 func TestStdAllocator(t *testing.T) {
 	slice := StdAllocator.Alloc(20)
-	if len(slice) != 0 {
-		t.Error("length not match")
-	}
-
-	if cap(slice) != 20 {
-		t.Error("cap not match")
-	}
+	assert.Equal(t, 0, len(slice), "length not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:46]:
```suggestion
	assert.Len(t, slice, 0)
```

File: util/arena/arena_test.go [hunk 11]
```diff
@@ -16,73 +16,37 @@ package arena
...
+	assert.Equal(t, 20, cap(slice), "cap not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:47]:
```suggestion
	assert.Equal(t, 20, cap(slice))
```

File: util/arena/arena_test.go [hunk 12]
```diff
@@ -16,73 +16,37 @@ package arena
...
 
 	slice = StdAllocator.AllocWithLen(10, 20)
-	if len(slice) != 10 {
-		t.Error("length not match")
-	}
-
-	if cap(slice) != 20 {
-		t.Error("cap not match")
-	}
+	assert.Equal(t, 10, len(slice), "length not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:50]:
```suggestion
	assert.Len(t, slice, 10)
```

File: util/arena/arena_test.go [hunk 13]
```diff
@@ -16,73 +16,37 @@ package arena
...
+	assert.Equal(t, 20, cap(slice), "cap not match")
```
Comment (tisonkun) [inline util/arena/arena_test.go:51]:
```suggestion
	assert.Equal(t, 20, cap(slice))
```

**Review by tisonkun — COMMENTED** [review]
@Ranxy thanks for your contribution! The migration is valid but we can do a bit better to polish the assertions. See my comments inline.

We don't have to bring the tech debt or overhead always.

**Comment by Ranxy** [comment]
@tisonkun Quite right, as it should be, I will fix this.

**Comment by tisonkun** [comment]
> @tisonkun Quite right, as it should be, I will fix this.

@Ranxy you can make use of GitHub suggestion comment UI instead of modifying manually.

See also https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

**Comment by Ranxy** [comment]
> > @tisonkun Quite right, as it should be, I will fix this.
> 
> @Ranxy you can make use of GitHub suggestion comment UI instead of modifying manually.
> 
> See also https://docs.github.com/en/github/collaborating-with-pull-requests/reviewing-changes-in-pull-requests/incorporating-feedback-in-your-pull-request#applying-suggested-changes

This is great, but I have an idea to use constants instead of magic numbers, what do you think?

**Comment by tisonkun** [comment]
@Ranxy you can push a commit and let's see the result. I think in test it does no harm you write the expected value as literals. Define a constant to use once may be redundant. Think of

```go
assert.Equal(t, 10, actual)

// and

expected := 10 // not unused anywhere else
assert.Equal(t. expected, actual)
```

But if the expected value is calculated or generated, that would be nice. It's up to taste, and you are always free to make a try.

**Review by tisonkun — APPROVED** [review]
Oops. I know what do you mean by consts now. It looks good.

+1 to merge.

**Review by disksing — APPROVED** [review]

File: util/arena/arena_test.go [hunk 14]
```diff
@@ -16,73 +16,50 @@ package arena
...
-	TestingT(t)
-}
+const (
+	arenaCap       = 1000
+	allocCapSmall  = 10
+	allocCapMedium = 20
+	allocCapOut    = 1024
+)
 
 func TestSimpleArenaAllocator(t *testing.T) {
```
Comment (tisonkun) [inline util/arena/arena_test.go:29]:
```suggestion
func TestSimpleArenaAllocator(t *testing.T) {
    t.Parallel()
```

File: util/arena/arena_test.go [hunk 15]
```diff
@@ -16,73 +16,50 @@ package arena
...
 
 	arena.Reset()
-	if arena.off != 0 || cap(arena.arena) != 1000 {
-		t.Error("off or cap not match")
-	}
+	assert.Zero(t, arena.off)
+	assert.Equal(t, arenaCap, cap(arena.arena))
 }
 
 func TestStdAllocator(t *testing.T) {
```
Comment (tisonkun) [inline util/arena/arena_test.go:57]:
```suggestion
func TestStdAllocator(t *testing.T) {
    t.Parallel()
```

**Review by tisonkun — COMMENTED** [review]
I think these tests can be run in parallel.

**Review by xhebox — APPROVED** [review]

**Comment by Ranxy** [comment]
/assign @disksing @xhebox 
Thanks for your review! May you help with merging this PR?

**Comment by Ranxy** [comment]
@disksing May you help with remerging this PR?

**Comment by tisonkun** [comment]
@Ranxy I think this PR is in merging phase. Please be patient for the bot merging it. There are several PRs merging concurrently in this repo every day.

**Comment by tisonkun** [comment]
/merge

If the bot doesn't work well for a while, you can comment and our infra team will take a look. You may at zhouqiang-cl or Mini256 .

**Comment by Ranxy** [comment]
@tisonkun Ok, I was under the mistaken impression that I needed to re-request merge after fetching upstream.

**Comment by tisonkun** [comment]
> @tisonkun Ok, I was under the mistaken impression that I needed to re-request merge after fetching upstream.

Generally the bot will merge master for you and you need not to do anything after `status/can-merge` tagged, although I don't like automation a bit since there is always corner case..

**Pull request merged**